contracts/compound-rate-keeper/AbstractCompoundRateKeeper.sol

```diff
@@ -209,4 +207,26 @@ abstract contract AbstractCompoundRateKeeper is ICompoundRateKeeper, Initializab
     function _getMaxRate() private pure returns (uint256) {
         return type(uint128).max * PRECISION;
     }
+
+    /**
+     * @notice Implementation of exponentiation by squaring
+     * @param base_ base, decimal number
+     * @param exponent_ exponent of x
+     * @param precision_ precision of x
+     * @return result_ result of exponentiation
+     */
+    function _rpow(
+        uint256 base_,
+        uint256 exponent_,
+        uint256 precision_
+    ) private pure returns (uint256 result_) {
+        result_ = exponent_ % 2 == 0 ? precision_ : base_;
```

Comment:
We can get rid of this ternary operator and replace a for loop with a while loop